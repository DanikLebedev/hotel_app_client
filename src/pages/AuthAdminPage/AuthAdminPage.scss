.auth-admin {
  width: 100%;
  background: url("../../assets/images/Chicago_70230862_1050x590_tcm-33720.jpg") center center / cover;
  position: fixed;
  top: 0;
  left: 0;
  z-index: -3;
  filter: brightness(0.5);
  min-height: 100%;
  &-wrapper {
    width: 100%;
    margin: 0 auto;
    display: flex;
    align-items: center;
    justify-content: space-around;
    min-height: 80vh;
  }

}

.auth-admin-form {
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: space-around;
  color: #000000;
  background: #ffffff;
  border-radius: 30px;
  flex-direction: column;
  padding: 30px;
  input {
    width: 200px;
    padding: 10px;
    border: none;
    border-bottom: 2px solid #cdcdcd;
    margin: 20px 20px 20px 0;
    outline: none;
    font-size: 1em;
    transition: .3s;
  }
}