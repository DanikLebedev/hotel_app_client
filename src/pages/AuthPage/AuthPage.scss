.wrapper {
  width: 100%;
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: space-around;
  min-height: 80vh;
}

.validate-error {
  animation: slideIn .5s;
  color: red;
}



@keyframes swipeOut {
  from {
    transform: translateY(0);
    opacity: 1;
  }
  to {
    transform: translateY(-200px);
    opacity: 0;
  }
}

.swipe-out {
  animation: swipeOut .5s ease-in-out !important;
}


.auth {
  background: url("../../assets/images/bg-01.jpg") no-repeat center center / cover;
  min-height: 100%;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  z-index: -3;
  filter: brightness(0.5);
}

.change-form-button {
  background: transparent;
  color: #1DA1F2;
  border: none;
}

.auth__wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  z-index: 3;
  background: #ffffff;
  padding: 40px;
  border-radius: 20px;
  text-align: center;
  margin-bottom: 20px;

  .auth__input {
    width: 200px;
    padding: 10px;
    border: none;
    border-bottom: 2px solid #cdcdcd;
    margin: 20px 20px 20px 0;
    outline: none;
    font-size: 16px;
  }

}

.btn__wrapper {
  display: flex;
  justify-content: space-around;

  .auth__btn {
    margin-right: 10px;
    text-decoration: none;
    padding: 5px 15px;
    background: #000;
    color: #ffffff;
    text-transform: uppercase;
    font-size: 16px;
  }
}

.signin-form {
  animation: slideInRight .5s;

}

.login-form {
  animation: slideInLeft .5s;
}