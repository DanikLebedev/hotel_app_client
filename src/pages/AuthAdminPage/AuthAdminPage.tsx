import React, { useContext, useState } from 'react';
import toaster from 'toasted-notes';
import './AuthAdminPage.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUser } from '@fortawesome/free-solid-svg-icons';
import { faLock } from '@fortawesome/free-solid-svg-icons/faLock';
import { useHistory } from 'react-router-dom';
import { AuthService } from '../../APIServices/authService';
import { UserData } from '../../interfaces/clientInterfaces';
import { ErrorMessage } from '../../components/ErrorsComponents/ErrorMessage';
import { useForm } from 'react-hook-form';
import Loader from '../../components/Loader/Loader';
import { ClientContext } from '../../context/client.context';

type InputEvent = React.ChangeEvent<HTMLInputElement>;

type AdminFormData = {
    email: string;
    password: string;
};

export const AuthAdminPage = () => {
    const { register, handleSubmit, errors } = useForm<AdminFormData>();
    const auth: ClientContext = useContext(ClientContext);
    const [form, setForm] = useState({ email: '', password: '' });
    const [loading, setLoading] = useState<boolean>(false);
    const history = useHistory();

    const changeHandler = (event: InputEvent): void => {
        setForm({ ...form, [event.target.name]: event.target.value });
    };

    const loginHandler = async (): Promise<void> => {
        try {
            setLoading(true);
            const data: UserData = await AuthService.loginAdmin(form, {
                'Content-Type': 'application/json',
            });
            auth.loginUser(data.token, data.userId, data.status, data.email);
            setLoading(false);
            if (data.status === 'admin') {
                history.push('/admin/info');
            } else if (data.status === 'manager') {
                history.push('/');
            }
            toaster.notify(data.message, {
                duration: 2000,
            });
        } catch (e) {
            console.log(e);
        }
    };

    return (
        <>
            <div className="auth-admin" />
            <div className="auth-admin-wrapper">
                {loading ? (
                    <Loader />
                ) : (
                    <div className="auth-admin-form">
                        <h1 className="text-center">Admin Login</h1>
                        <label>
                            <FontAwesomeIcon icon={faUser} />
                            <input
                                className="auth__input"
                                type="text"
                                name="email"
                                id="email"
                                placeholder="e.g asd@mail.ru"
                                ref={register({ required: true, pattern: /^\S+@\S+$/i })}
                                onChange={changeHandler}
                            />
                        </label>
                        <ErrorMessage error={errors.email} type={'error'} />
                        <label>
                            <FontAwesomeIcon icon={faLock} />
                            <input
                                className="auth__input"
                                type="password"
                                name="password"
                                id="password"
                                placeholder="password"
                                ref={register({ required: true, minLength: 6 })}
                                onChange={changeHandler}
                            />
                        </label>
                        <ErrorMessage error={errors.password} type={'error'} />

                        <button className="button btn-black" onClick={handleSubmit(loginHandler)}>
                            Login
                        </button>
                    </div>
                )}
            </div>
        </>
    );
};
