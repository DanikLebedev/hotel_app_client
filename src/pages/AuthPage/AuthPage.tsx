import React, { useContext, useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUser, faEdit, faUserEdit } from '@fortawesome/free-solid-svg-icons';
import { faLock } from '@fortawesome/free-solid-svg-icons/faLock';
import toaster from 'toasted-notes';
import 'toasted-notes/src/styles.css';
import './AuthPage.scss';
import Loader from '../../components/Loader/Loader';
import { ClientContext } from '../../context/client.context';
import { useHistory } from 'react-router-dom';
import { Data, LoginData, RegisterData, UserData } from '../../interfaces/clientInterfaces';
import { AuthService } from '../../APIServices/authService';
import { useForm } from 'react-hook-form';
import { ErrorMessage } from '../../components/ErrorsComponents/ErrorMessage';

type InputEvent = React.ChangeEvent<HTMLInputElement>;

type FormData = {
    email: string;
    password: string;
    name?: string;
    lastName?: string;
};

const AuthPage: React.FC = () => {
    const [loading, setLoading] = useState<boolean>(false);
    const history = useHistory();
    const auth: ClientContext = useContext(ClientContext);
    const [haveAccount, setHaveAccount] = useState<boolean>(true);
    const [resetForm, setResetForm] = useState({ email: '' });
    const [forgotPassword, setForgotPassword] = useState(false);

    const [form, setForm] = useState<LoginData>({ email: '', password: '' });
    const [registerForm, setRegisterForm] = useState<RegisterData>({ email: '', password: '', name: '', lastName: '' });

    const { register, handleSubmit, errors } = useForm<FormData>();

    const changeResetHandler = (event: InputEvent): void => {
        setResetForm({ ...resetForm, [event.target.name]: event.target.value });
    };

    const resetHandler = async (): Promise<void> => {
        const response = await fetch('/api/client/reset', {
            method: 'POST',
            body: JSON.stringify(resetForm),
            headers: {
                'Content-Type': 'application/json',
            },
        });
        const data: Data = await response.json();
        setForgotPassword(false);
        toaster.notify(data.message, {
            duration: 2000,
        });
    };

    const changeHandler = (event: InputEvent): void => {
        setForm({ ...form, [event.target.name]: event.target.value });
    };

    const changeRegisterInputHandler = (event: InputEvent): void => {
        setRegisterForm({ ...registerForm, [event.target.name]: event.target.value });
    };

    const registerHandler = async (): Promise<void> => {
        try {
            setLoading(true);
            const data: UserData = await AuthService.registerUser(registerForm, {
                'Content-Type': 'application/json',
            });
            toaster.notify(data.message, {
                duration: 2000,
            });
            const loginData: UserData = await AuthService.loginUser(registerForm, {
                'Content-Type': 'application/json',
            });
            setLoading(false);
            auth.loginUser(loginData.token, loginData.userId, loginData.status, loginData.email);
            history.push('/');
        } catch (e) {}
    };

    const loginHandler = async (): Promise<void> => {
        try {
            setLoading(true);
            const data: UserData = await AuthService.loginUser(form, {
                'Content-Type': 'application/json',
            });
            setLoading(false);
            if (!data.token) {
                toaster.notify(data.message, {
                    duration: 2000,
                });
            } else {
                auth.loginUser(data.token, data.userId, data.status, data.email);
                history.push('/');
            }
        } catch (e) {}
    };

    const loginForm: JSX.Element = (
        <div className={haveAccount ? 'auth__wrapper login-form' : 'auth__wrapper login-form swipe-out'}>
            <h1>Log in</h1>
            <button className="change-form-button" onClick={() => setHaveAccount(false)}>
                Don&apos;t have an account?
            </button>
            <label>
                <FontAwesomeIcon icon={faUser} />
                <input
                    className="auth__input"
                    type="text"
                    name="email"
                    id="email"
                    placeholder="e.g asd@mail.ru"
                    onChange={changeHandler}
                    ref={register({ required: true, pattern: /^\S+@\S+$/i })}
                />
            </label>
            <ErrorMessage error={errors.email} type={'error'} />
            <label>
                <FontAwesomeIcon icon={faLock} />
                <input
                    className="auth__input"
                    type="password"
                    name="password"
                    id="password"
                    placeholder="password"
                    onChange={changeHandler}
                    ref={register({ required: true, minLength: 6 })}
                />
            </label>
            <ErrorMessage error={errors.password} type={'error'} />
            <button className="change-form-button mb-3" onClick={() => setForgotPassword(true)}>
                Reset password
            </button>
            <div className="btn__wrapper">
                <button className="button btn-black" onClick={handleSubmit(loginHandler)}>
                    Login
                </button>
            </div>
        </div>
    );

    const forgotPasswordForm: JSX.Element = (
        <div className="auth__wrapper signin-form">
            <h1>Enter your email </h1>
            <button className="change-form-button" onClick={() => setForgotPassword(false)}>
                Go back
            </button>
            <label>
                <FontAwesomeIcon icon={faUser} />
                <input
                    className="auth__input"
                    type="text"
                    name="email"
                    id="email"
                    placeholder="e.g asd@mail.ru"
                    onChange={changeResetHandler}
                    ref={register({ required: true, pattern: /^\S+@\S+$/i })}
                />
            </label>

            <div className="btn__wrapper">
                <button className="button btn-black" onClick={handleSubmit(resetHandler)}>
                    Send email
                </button>
            </div>
        </div>
    );

    const signInForm: JSX.Element = (
        <div className="auth__wrapper signin-form">
            <h1>Sign in</h1>
            <label>
                <FontAwesomeIcon icon={faUser} />
                <input
                    className="auth__input"
                    type="text"
                    name="email"
                    id="email"
                    placeholder="e.g asd@mail.ru"
                    onChange={changeRegisterInputHandler}
                    ref={register({ required: true, pattern: /^\S+@\S+$/i })}
                />
            </label>
            <ErrorMessage error={errors.email} type={'error'} />
            <label>
                <FontAwesomeIcon icon={faLock} />
                <input
                    className="auth__input"
                    type="password"
                    name="password"
                    id="password"
                    placeholder="password"
                    onChange={changeRegisterInputHandler}
                    ref={register({ required: true, minLength: 6 })}
                />
            </label>
            <ErrorMessage error={errors.password} type={'error'} />
            <label>
                <FontAwesomeIcon icon={faUserEdit} />
                <input
                    className="auth__input"
                    type="text"
                    name="name"
                    id="name"
                    placeholder="your name"
                    onChange={changeRegisterInputHandler}
                    ref={register({ required: true })}
                />
            </label>
            <ErrorMessage error={errors.name} type={'error'} />
            <label>
                <FontAwesomeIcon icon={faEdit} />
                <input
                    className="auth__input"
                    type="text"
                    name="lastName"
                    id="lastName"
                    placeholder="your last name"
                    onChange={changeRegisterInputHandler}
                    ref={register({ required: true })}
                />
            </label>
            <ErrorMessage error={errors.lastName} type={'error'} />
            <button className="change-form-button" onClick={() => setHaveAccount(true)}>
                Have an account?
            </button>
            <div className="btn__wrapper">
                <button className="button btn-black" onClick={handleSubmit(registerHandler)}>
                    Register
                </button>
            </div>
        </div>
    );
    return (
        <>
            <div className="auth" />
            <div className="wrapper">
                {loading ? <Loader /> : forgotPassword ? forgotPasswordForm : haveAccount ? loginForm : signInForm}
            </div>
        </>
    );
};

export default AuthPage;
