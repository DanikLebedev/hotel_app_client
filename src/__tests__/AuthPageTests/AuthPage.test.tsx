import { shallow } from 'enzyme';
import React from 'react';
import AuthPage from '../../pages/AuthPage/AuthPage';
import { AuthAdminPage } from '../../pages/AuthAdminPage/AuthAdminPage';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

it('should renders without crashing', function() {
    const wrapper = shallow(
        <Router history={createMemoryHistory()}>
            <AuthPage />
        </Router>,
    );
    expect(wrapper).toMatchSnapshot();
});

it('should renders without crashing', function() {
    const wrapper = shallow(
        <Router history={createMemoryHistory()}>
            <AuthAdminPage />
        </Router>,
    );
    expect(wrapper).toMatchSnapshot();
});
