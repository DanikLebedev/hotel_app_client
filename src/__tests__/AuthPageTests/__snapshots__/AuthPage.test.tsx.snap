// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "pz1y7j",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "pz1y7j",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "location": Object {
        "hash": "",
        "key": "pz1y7j",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "match": Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <AuthPage />
</ContextProvider>
`;

exports[`should renders without crashing 2`] = `
<ContextProvider
  value={
    Object {
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "8n7q4a",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "8n7q4a",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "location": Object {
        "hash": "",
        "key": "8n7q4a",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "match": Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <AuthAdminPage />
</ContextProvider>
`;
